asyncio/event.py

```diff
@@ -62,7 +62,7 @@ async def wait(self):
             self.waiting.push_head(core.cur_task)
             # Set calling task's data to the event's queue so it can be removed if needed
             core.cur_task.data = self.waiting
-            await core.sleep(0)
+            yield
```

Comment:
I changed this to `await core.sleep(0)` to get it to be standard Python. I don't remember if I did this because pylint or black complained, or because it actually didn'twork with our implementation of async/await, which does not use `yield` the way MicroPython does

If `port/unix` is using the micropython async/await, then that is an incompatibility, and we should switch that version to use our async/await, which uses `__await__()`, not `__iter()__`, etc. I had to define `__await__()` in our adaptation of asyncio.

So I don't think this should do this. There some discussion about `yield` vs `sleep`, and `__iter__()` vs `__await__()` here:
https://github.com/micropython/micropython/issues/8508

Reply:
Also see in particular
https://github.com/micropython/micropython/issues/8508#issuecomment-1161047942
and
https://github.com/peterhinch/micropython-async/blob/master/v3/docs/TUTORIAL.md#41-awaitable-classes,
mentioned in https://github.com/micropython/micropython/issues/8508#issuecomment-1161564954

Reply:
I tried running `tests/extmod/uasyncio_events.py` before and after this change.

After this change, the test passes both in the Unix test harness and (just based on visual verification of the output, not rigorous diffing) on the device.

Before this change, both tests fail in a similar way:
```
Traceback (most recent call last):
  File "asyncio/core.py", line 211, in run_until_complete
StopIteration: 
Traceback (most recent call last):
  File "code.py", line 1, in <module>
  File "uasyncio_event.py", line 98, in <module>
  File "asyncio/core.py", line 262, in run
  File "asyncio/core.py", line 252, in run_until_complete
  File "asyncio/core.py", line 340, in call_exception_handler
  File "asyncio/core.py", line 333, in default_exception_handler
AttributeError: 'NoneType' object has no attribute '__traceback__'
```

The same goes for the change to `lock.py` and the `asyncio_lock.py` test in the core.

I recognize that it is not standard Python to write `yield` here. However, the current replacement is not equivalent in functionality in CircuitPython, so either we need to `yield` or we need to find the equivalent replacement that is also accepted as standard Python.